libraries/anvill_passes/src/RecoverStackFrameInformation.cpp

```diff
@@ -274,18 +300,51 @@ RecoverStackFrameInformation::UpdateFunction(
   // Pre-initialize the stack frame to zero if we have been requested
   // to do so. From the whole FunctionPass class, we get the setting
   // from the LiftingOptions class
-  if (init_strategy == StackFrameStructureInitializationProcedure::kZeroes) {
-    auto null_value = llvm::Constant::getNullValue(stack_frame_type);
-    builder.CreateStore(null_value, stack_frame_alloca);
+  switch (init_strategy) {
+    case StackFrameStructureInitializationProcedure::kZeroes: {
+      auto null_value = llvm::Constant::getNullValue(stack_frame_type);
+      builder.CreateStore(null_value, stack_frame_alloca);
 
-  } else if (init_strategy ==
-             StackFrameStructureInitializationProcedure::kUndef) {
-    auto null_value = llvm::UndefValue::get(stack_frame_type);
-    builder.CreateStore(null_value, stack_frame_alloca);
+      break;
+    }
 
-  } else if (init_strategy !=
-             StackFrameStructureInitializationProcedure::kNone) {
-    return StackAnalysisErrorCode::InvalidParameter;
+    case StackFrameStructureInitializationProcedure::kUndef: {
+      auto null_value = llvm::UndefValue::get(stack_frame_type);
+      builder.CreateStore(null_value, stack_frame_alloca);
+
+      break;
+    }
+
+    case StackFrameStructureInitializationProcedure::kSymbolic: {
+      auto current_offset = stack_frame_analysis.lowest_offset;
+      auto &module = *function.getParent();
+      auto byte_type = llvm::Type::getInt8Ty(module.getContext());
+
+      for (auto i = 0U; i < stack_frame_analysis.size; ++i) {
+        auto stack_frame_byte = builder.CreateGEP(
+            stack_frame_alloca,
+            {builder.getInt32(0), builder.getInt32(0), builder.getInt32(i)});
+
+        auto symbolic_value_res =
+            GetStackSymbolicByteValue(module, current_offset);
+
+        if (!symbolic_value_res.Succeeded()) {
+          return symbolic_value_res.TakeError();
+        }
+
+        auto symbolic_value = symbolic_value_res.TakeValue();
+        ++current_offset;
+
+        auto casted_sym_value =
+            llvm::ConstantExpr::getPtrToInt(symbolic_value, byte_type);
+
+        builder.CreateStore(casted_sym_value, stack_frame_byte);
+      }
+
+      break;
+    }
+
+    case StackFrameStructureInitializationProcedure::kNone: break;
```

Comment:
Comment that this may or may not be equivalent to storing undefined, and it depends on compiler switches and compiler version behaviour. Loaded values from uninitialized memory are likely implicitly undefined, whereas above, with `kUndef`, they load explicitly undefined.

Reply:
I've added a "look at the enum definition" comment here and documented the strategies in more details in the public interface